src/gui/models/rangeModel.cpp

```diff
@@ -0,0 +1,178 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+// Copyright (c) 2023 Team Dissolve and contributors
+
+#include "gui/models/rangeModel.h"
+#include <QBrush>
+#include <fmt/format.h>
+
+// set range data
+void RangeModel::setData(std::vector<Range> &ranges)
+{
+
+    beginResetModel();
+    ranges_ = ranges;
+    endResetModel();
+}
+
+Range *RangeModel::getRange(const QModelIndex &index) const { return &ranges_->get()[index.row()]; }
```

Comment:
May as well return a reference here, no? But you might want to add a few other checks (e.g. index.isValid()) to try and catch the obvious segfault.
```suggestion
Range &RangeModel::getRange(const QModelIndex &index) const { return ranges_->get()[index.row()]; }
```